qiskit/algorithms/optimizers/adam_amsgrad.py

```diff
@@ -114,6 +114,13 @@ def __init__(
                 writer = csv.DictWriter(csv_file, fieldnames=fieldnames)
                 writer.writeheader()
 
+    @property
+    def settings(self) -> Dict[str, Any]:
+        settings = {"name": "ADAM"}
+        settings.update(self._options)
```

Comment:
Given `set_options()` allows setting arbitrary options, I think it'd be safer to hard code the known option keywords rather than returning `self._options`.

Reply:
done 👍🏻